LibraryManagement.cpp

```diff
@@ -268,76 +300,75 @@ void Bookmark::editUser(int userCounter) {
 	main();
 	}
 
-void Bookmark::viewAllUsers(int userCounter) {
+void Bookmark::viewAllUsers() {
     cout << "VIEW ALL USERS\n\n";
     if(bookMark.userCounter==0) {
 			cout<<"There are currently no users added."<<endl;
 			main();
 		}
-    for (int i = 0; i < userCounter; i++) {
+    for (int i = 0; i < bookMark.userCounter; i++) {
         cout<<"User details\n\n";
         cout << "First name: " << bookMark.users[i].getFirstName() << endl;
         cout << "Last name: " << bookMark.users[i].getLastName() << endl;
         cout << "ID: " << bookMark.users[i].getID() << endl;
 
 	}
-	main();
+	menu.displayMainMenu();
 }
-void Bookmark::deleteUser(int userCounter) {
+void Bookmark::deleteUser() {
     string ID;
     int choice;
-    	cout << "DELETE USER\n\n";
-    if(userCounter==0){
-		cout<<"There are no users to delete!";
-		main();
-	}
-    	cout << "Enter ID: ";
-    	cin >> ID;
+    bool found = false;
 
+    cout << "DELETE USER\n\n";
 
-for (int i = 0; i < userCounter; i++) {
-    if (bookMark.users[i].getID() == ID) {
-        cout<<"\nUser found\n\n";
-        cout<<"Do you want delete?\n[1]Yes\n[2]No\n\nEnter choice: ";
-        cin>>choice;
-        if(choice==1) {
-        	bookMark.users[i].setFirstName("");
-        	bookMark.users[i].setLastName("");
-        	bookMark.users[i].setID("");
-        	for(int j=i;j<userCounter;j++) {
-        		bookMark.users[j]=bookMark.users[j+1];
-			}
-		    bookMark.users[9].setFirstName("");
-			bookMark.users[9].setLastName("");
-			bookMark.users[9].setID("");
-			bookMark.decrementUser(bookMark.userCounter);
-			cout<<"User successfully deleted.";
-			main();
-		}
-		else {
-			main();
-		}
+    if (userCounter == 0) {
+        cout << "There are no users to delete!" << endl;
+        return;
     }
-}
-	cout<<"\nUser not found!";
-	main();
 
+    cout << "Enter ID: ";
+    cin >> ID;
 
+    for (int i = 0; i < userCounter; i++) {
+        if (bookMark.users[i].getID() == ID) {
```

Comment:
you can create findUserById method inside bookmark and extract this logic to that method. It would make code more readable